flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/codegen/sort/ComparatorCodeGenerator.scala

```diff
@@ -36,25 +37,27 @@ object ComparatorCodeGenerator {
     * @param conf        Table config.
     * @param name        Class name of the function.
     *                    Does not need to be unique but has to be a valid Java class identifier.
-    * @param keys        key positions describe which fields are keys in what order.
-    * @param keyTypes    types for the key fields, in the same order as the key fields.
-    * @param orders      sorting orders for the key fields.
-    * @param nullsIsLast Ordering of nulls.
+    * @param input        input type.
+    * @param sortSpec     sort specification.
     * @return A GeneratedRecordComparator
     */
   def gen(
       conf: TableConfig,
       name: String,
-      keys: Array[Int],
-      keyTypes: Array[LogicalType],
-      orders: Array[Boolean],
-      nullsIsLast: Array[Boolean]): GeneratedRecordComparator = {
+      input: RowType,
+      sortSpec: SortSpec): GeneratedRecordComparator = {
     val className = newName(name)
     val baseClass = classOf[RecordComparator]
 
     val ctx = new CodeGeneratorContext(conf)
     val compareCode = GenerateUtils.generateRowCompare(
-      ctx, keys, keyTypes, orders, nullsIsLast, "o1", "o2")
+        ctx,
+        sortSpec.getFieldIndices,
+        sortSpec.getFieldTypes(input),
+        sortSpec.getAscendingOrders,
+        sortSpec.getNullsIsLast,
```

Comment:
replace the sort related parameters with `SortSpec` for `GenerateUtils.generateRowCompare` method ?